llama_index/llms/vllm.py

```diff
@@ -186,6 +186,14 @@ def __init__(
             output_parser=output_parser,
         )
 
+    def __del__(self):
+        from vllm.model_executor.parallel_utils.parallel_state import destroy_model_parallel
+        import torch
+        destroy_model_parallel()
```

Comment:
will it destroy one of the parallel models?

Reply:
This will destroy `self._client` which is the `VLLM` model.

Reply:
@hatianzhang 

```python
def destroy_model_parallel():
    """Set the groups to none and destroy them."""
    global _TENSOR_MODEL_PARALLEL_GROUP
    if _TENSOR_MODEL_PARALLEL_GROUP:
        torch.distributed.destroy_process_group(_TENSOR_MODEL_PARALLEL_GROUP)
    _TENSOR_MODEL_PARALLEL_GROUP = None
    global _PIPELINE_MODEL_PARALLEL_GROUP
    if _PIPELINE_MODEL_PARALLEL_GROUP:
        torch.distributed.destroy_process_group(_PIPELINE_MODEL_PARALLEL_GROUP)
    _PIPELINE_MODEL_PARALLEL_GROUP = None
    global _PIPELINE_GLOBAL_RANKS
    _PIPELINE_GLOBAL_RANKS = None
```

Here is the code from vllm. 
https://github.com/vllm-project/vllm/blob/main/vllm/model_executor/parallel_utils/parallel_state.py

It looks they save tensor parallel group at global variable when running. 
For using multiple vllm instance, one have to be destroyed.